src/dsql/DsqlRequests.cpp

```diff
@@ -177,6 +177,9 @@ void DsqlRequest::destroy(thread_db* tdbb, DsqlRequest* dsqlRequest)
 	{
 		childStatement->addFlags(DsqlStatement::FLAG_ORPHAN);
 		childStatement->setParentRequest(nullptr);
+		childStatement->setParentDbKey(nullptr);
+		childStatement->setParentRecVersion(nullptr);
```

Comment:
Isn't it pointless to clean fields of statement which is going to be removed from cache and thus destructed? FLAG_ORPHAN already prevent any usage of the statement.